### PR TITLE
Ndev 3232 run tracer tests on geth

### PR DESCRIPTION
**Pull request by romanova-natasha**
Tracer tests were run on Geth.
Bugs were created: NDEV-3248, NDEV-3249, NDEV-3250, NDEV-3251, NDEV-3252, NDEV-3260, NDEV-3261.
Tracer tests are marked according to the bugs.

Changes from NDEV-2981-traceCall-overrides-param have been merged into this branch, all override params tests are skipped before this feature implemented.

### CODE REVIEW

File: integration/tests/tracer/test_tracer_debug_trace_transactions_callTracer.py [hunk 1]
```diff
@@ -107,9 +126,9 @@ def test_callTracer_type_create(self, storage_contract_with_deploy_tx):
         expected_response["gasUsed"] =  hex(receipt["gasUsed"])
         expected_response["type"] = "CREATE"
 
-        self.assert_response_contains_expected(expected_response, response)
+        self.assert_response_contains_expected(pytestconfig, expected_response, response)
```
Comment (kristinaNikolaevaa) [inline integration/tests/tracer/test_tracer_debug_trace_transactions_callTracer.py:129]: maybe it's better just skip this check for geth network?

File: integration/tests/tracer/test_tracer_debug_trace_transactions_callTracer.py [hunk 2]
```diff
@@ -349,11 +381,14 @@ def test_callTracer_call_contract_from_other_contract_revert(self, events_checke
         response = self.tracer_api.send_rpc_and_wait_response("debug_traceTransaction", params)
 
         address_to = instruction_tx["to"].lower()
-        reason = f"Insufficient balance for transfer, account = {address_to}, chain = 111, required = 1"
-        expected_response = self.fill_expected_response(instruction_tx, receipt, logs=True, revert=True, revert_reason=reason)
-        self.assert_response_contains_expected(expected_response, response)
+        if pytestconfig.getoption("--network") == "geth":
+            reason = "insufficient balance for transfer"
+        else:
```
Comment (kristinaNikolaevaa) [inline integration/tests/tracer/test_tracer_debug_trace_transactions_callTracer.py:386]: so, it's expected here that responses are different?

File: utils/helpers.py [hunk 3]
```diff
@@ -183,4 +183,7 @@ def serialize_instruction(program_id, instruction) -> bytes:
         serialized += key.is_writable.to_bytes(1, "little")
 
     serialized += len(instruction.data).to_bytes(8, "little") + instruction.data
-    return serialized
\ No newline at end of file
+    return serialized
+
+def padhex(s, size):
```
Comment (kristinaNikolaevaa) [inline utils/helpers.py:188]: move it to helpers please

File: utils/web3client.py [hunk 4]
```diff
@@ -16,6 +16,7 @@
 from utils import helpers
 from utils.consts import InputTestConstants, Unit
 from utils.helpers import decode_function_signature
+from utils.helpers import wait_condition
```
Comment (kristinaNikolaevaa) [inline utils/web3client.py]: it can be deleted

File: integration/tests/tracer/test_tracer_debug_trace_transactions_callTracer.py (hunk 2 above)
Comment (romanova-natasha) [reply integration/tests/tracer/test_tracer_debug_trace_transactions_callTracer.py:386]: we have same message but with more details
"Insufficient balance for transfer, account = {address_to}, chain = {self.web3_client.eth.chain_id}, required = 1"

File: utils/helpers.py (hunk 3 above)
Comment (romanova-natasha) [reply utils/helpers.py:188]: but it is already in utils/helpers.py

File: utils/web3client.py (hunk 4 above)
Comment (romanova-natasha) [reply utils/web3client.py]: done

File: integration/tests/tracer/test_tracer_debug_trace_transactions_callTracer.py (hunk 1 above)
Comment (romanova-natasha) [reply integration/tests/tracer/test_tracer_debug_trace_transactions_callTracer.py:129]: I already do it. I added network check to skip some of fields when run tests on geth

```
if pytestconfig.getoption("--network") == "geth":
            # we do not fill whole response, that is why we skip some of fields
            # we can build compare function for each field in the future if it needed
            exclude_list = ["root['gas']", "root['output']", "root['value']"]
            if "calls" in expected_response:
                for i in range(len(expected_response["calls"])):
                    exclude_list.append(f"root['calls'][{i}]['to']")
                    exclude_list.append(f"root['calls'][{i}]['gas']")
                    exclude_list.append(f"root['calls'][{i}]['gasUsed']")
                    exclude_list.append(f"root['calls'][{i}]['input']")
                    exclude_list.append(f"root['calls'][{i}]['output']")
                    exclude_list.append(f"root['calls'][{i}]['value']")
                    exclude_list.append(f"root['calls'][{i}]['logs'][0]['address']")
                    exclude_list.append(f"root['calls'][{i}]['logs'][0]['position']")
        else:
            exclude_list = []
```

**Review by kristinaNikolaevaa — APPROVED** [review]

**Pull request merged**